### PR TITLE
fix: deprecate maxExtension in favour of maxExtensionMinutes

### LINKED ISSUE #1208: Default option `maxExtensionMinutes` is treated as seconds
The default options is set as 60 minutes:

https://github.com/googleapis/nodejs-pubsub/blob/ccebf72508ba9f410c539eedaa879ec6fa423c0b/src/default-options.ts#L29-L31

The value `60` is applied here:

https://github.com/googleapis/nodejs-pubsub/blob/74fc12e8d1e26939309748536bfecb751be0f079/src/lease-manager.ts#L188

However `maxExtension` is documented as being a value in seconds, defaulting to `60`:

https://github.com/googleapis/nodejs-pubsub/blob/74fc12e8d1e26939309748536bfecb751be0f079/src/lease-manager.ts#L39-L40

Arguably the default should be `1` and when that default is applied it needs to be modified by 60.

### PR DESCRIPTION
**Pull request by feywind**
Background: maxExtension was incorrectly interpreted as seconds in the LeaseManager. This means that if users were relying on client side lease extension in subscribers, they would not actually extend leases for 60 minutes, but only 60 seconds. Also if the user passed in values, they'd be interpreted incorrectly. This PR deprecates the old field in favour of a new one that's explicitly named like the defaults, but also converts the old value if needed.

Fixes: https://github.com/googleapis/nodejs-pubsub/issues/1208

### CODE REVIEW

**Review by anguillanneuf — APPROVED** [review]
LGTM. Looks like the Node.js [sample](https://cloud.google.com/pubsub/docs/pull#flow_control) just uses `maxMessages`. Can you think of any other places we need to update?

File: src/lease-manager.ts [hunk 1]
```diff
@@ -182,10 +185,19 @@ export class LeaseManager extends EventEmitter {
    * @private
    */
   setOptions(options: FlowControlOptions): void {
+    // Convert the old deprecated maxExtension to avoid breaking clients.
+    if (
+      options.maxExtension !== undefined &&
```
Comment (JustinBeckwith) [inline src/lease-manager.ts:190]: What happens if both are defined?

**Review by JustinBeckwith — APPROVED** [review]
LGTM with nit.

**Comment by feywind** [comment]
> LGTM. Looks like the Node.js [sample](https://cloud.google.com/pubsub/docs/pull#flow_control) just uses `maxMessages`. Can you think of any other places we need to update?

I think this is maxMessages rather than maxExtension, should be unchanged.

**Pull request merged**